app/script.rb

```diff
@@ -0,0 +1,188 @@
+      manual_login
+    else
+      local_login
+    end
+    browser.button(class: "wb-button").click
+    sleep 2
+    if browser.div(class: "block__cards-accounts").exist?
+      puts "Authentication successful"
+    else
+      puts "Authentication failed, try again"
+      manual_login
+    end
+  end
+
+  def local_login
+    file = File.read('data/login.json')
+    json = JSON.parse(file)
+    browser.text_field(name: "login").set(json["login"])
+    browser.text_field(name: "password").set(json["password"])
+  end
+
+  def manual_login
+    puts "Enter your login"
+    browser.text_field(name: "login").set(gets.chomp)
+    puts "Enter your password"
+    browser.text_field(name: "password").set(gets.chomp)
+    if browser.text_field(name: "captcha").present?
+      puts "Enter CAPTCHA"
+      browser.text_field(name: "captcha").set(gets.chomp)
+    end
+  end
+
+  def accounts_html
+    Nokogiri::HTML.parse(browser.div(class: "contracts-section").html)
+  end
+
+  def transactions_html
+    Nokogiri::HTML.parse(browser.div(class: "operations").html)
+  end
+
+  def check_accounts
+    browser.goto(ACCOUNTS_URL)
+    puts "Fetching account informatrion"
+    @accounts = Array.new
+    accounts_html.css('div.contracts-section').map do |page|
+      unless page.css('div.section-title.no-data-error').any?
+        name     = page.css('div.main-info').css('a.name').text
+        balance  = page.css('div.primary-balance').css('span.amount').first.text
+        currency = page.css('div.primary-balance').css('span.amount').last.text
+        nature   = page.css('div.section-title.h-small').text.downcase.capitalize
+        account  = Accounts.new(name, balance, currency, nature)
+        @accounts << account
+      end
+    end
+  end
+
+  def check_transactions
+    browser.goto(TRANSACTIONS_URL)
+    set_date
+    sleep 2 
+    puts "Fetching transactions for the last two months"
+    @transactions = Array.new
+    transactions_html.css('li.history-item.success').map do |page|
+      year  = page.xpath('../../preceding-sibling::div[@class = "month-delimiter"]').last.text.split[1]
+      month = page.xpath('../../preceding-sibling::div[@class = "month-delimiter"]').last.text.split[0]
+      day   = page.parent.parent.css('div.day-header').text.split[0]
+      time  = page.css('span.history-item-time').text
+      date  = (year + " " + month + " " + day).to_s
+      description = page.css('span.history-item-description').text.split.join(" ")
+      if !page.css('span.history-item-amount.transaction.income').text.empty?
+        amount = page.css('span.history-item-amount.transaction.income').text
+      elsif !page.css('span.history-item-amount.total').text.empty?
+        amount = page.css('span.history-item-amount.total').text
+      else
+        amount = page.css('span.history-item-amount.transaction').text
+      end
+    transaction = Transactions.new(date, description, amount)
+    @transactions << transaction
+    end
```

Comment:
Something wrong with the indentation. There is no possible way you there can be 2 lines of code one of which is `end` and they are still lined up.